src/trio/_tests/check_type_completeness.py

```diff
@@ -0,0 +1,221 @@
+"""This is a file that wraps calls to `pyright --verifytypes`, achieving two things:
+1. give an error if docstrings are missing.
+    pyright will give a number of missing docstrings, and error messages, but not exit with a non-zero value.
+2. filter out specific errors we don't care about.
+    this is largely due to 1, but also because Trio does some very complex stuff and --verifytypes has few to no ways of ignoring specific errors.
+"""
+from __future__ import annotations
+
+# this file is not run as part of the tests, instead it's run standalone from check.sh
+import argparse
+import json
+import subprocess
+import sys
+from pathlib import Path
+
+import trio
+import trio.testing
+
+# not needed if everything is working, but if somebody does something to generate
+# tons of errors, we can be nice and stop them from getting 3*tons of output
+printed_diagnostics: set[str] = set()
+
+
+# TODO: consider checking manually without `--ignoreexternal`, and/or
+# removing it from the below call later on.
+def run_pyright(platform: str) -> subprocess.CompletedProcess[bytes]:
+    return subprocess.run(
+        [
+            "pyright",
+            # Specify a platform and version to keep imported modules consistent.
+            f"--pythonplatform={platform}",
+            "--pythonversion=3.8",
+            "--verifytypes=trio",
+            "--outputjson",
+            "--ignoreexternal",
+        ],
+        capture_output=True,
+    )
+
+
+def has_docstring_at_runtime(name: str) -> bool:
+    """Pyright gives us an object identifier of xx.yy.zz
+    This function tries to decompose that into its constituent parts, such that we
+    can resolve it, in order to check whether it has a `__doc__` at runtime and
+    verifytypes misses it because we're doing overly fancy stuff.
+    """
+    assert trio.testing
+
+    # figure out what part of the name is the module, so we can "import" it
+    name_parts = name.split(".")
+    split_i = 1
+    if name_parts[1] == "tests":
+        return True
+    if name_parts[1] in ("_core", "testing"):  # noqa: SIM108
+        split_i = 2
+    else:
+        split_i = 1
+    module = sys.modules[".".join(name_parts[:split_i])]
+
+    # traverse down the remaining identifiers with getattr
+    obj = module
+    try:
+        for obj_name in name_parts[split_i:]:
+            obj = getattr(obj, obj_name)
+    except AttributeError as exc:
+        # asynciowrapper does funky getattr stuff
+        if "AsyncIOWrapper" in str(exc) or name in (
+            # Symbols not existing on all platforms, so we can't dynamically inspect them.
+            # Manually confirmed to have docstrings but pyright doesn't see them due to
+            # export shenanigans. TODO: actually manually confirm that.
+            # darwin
+            "trio.lowlevel.current_kqueue",
+            "trio.lowlevel.monitor_kevent",
+            "trio.lowlevel.wait_kevent",
+            "trio._core._io_kqueue._KqueueStatistics",
```

Comment:
Idea:

```suggestion
            # --- darwin
            "trio.lowlevel.current_kqueue",
            "trio.lowlevel.monitor_kevent",
            "trio.lowlevel.wait_kevent",
            "trio._core._io_kqueue._KqueueStatistics",
            # --- darwin
```

Then the test splits the file on `# --- darwin` (or whatever platform its running), takes the middle, `skipped_symbols = eval('[' + middle + ']')`, and check those.

Reply:
Hrm, this starts feeling very silly that it's separate from `test_static_tool_sees_class_members`.